plugin/admin/options/option-types/class-selectfield.php

```diff
@@ -37,28 +37,43 @@ public function register() {
 	 * @return int Sanitized value.
 	 */
 	public function sanitize( $value ) {
-		if ( 'pixels' === $value ) {
-			return 'pixels';
-		}
-
-		if ( 'percentage' === $value ) {
-			return 'percentage';
-		}
-
-		return $this->default_value;
+		return esc_html( $value );
 	}
 
 	/**
 	 * Display field for updating the option
 	 */
 	public function display() {
-		echo '<label for="sgdd-' . esc_attr( $this->id ) . '">
-						<input type="radio" id="sgdd-' . esc_attr( $this->id ) . '-list" name="' . esc_attr( $this->id ) . '" value="list" ' . ( $this->get() === 'list' ? 'checked' : '' ) . '> List
-					</label>
-					<br>
-					<label for="sgdd-' . esc_attr( $this->id ) . '">
-						<input type="radio" id="sgdd-' . esc_attr( $this->id ) . '-grid" name="' . esc_attr( $this->id ) . '" value="grid" ' . ( $this->get() === 'grid' ? 'checked' : '' ) . '> Grid
-					</label>
-		';
+		$display = '';
+		$inputs;
+
+		// Folder view type
+		if ( 'sgdd_folder_type' === $this->id ) {
+			$inputs = array(
+				array( 'list', __( 'List', 'skaut-google-drive-documents' ) ),
+				array( 'grid', __( 'Grid', 'skaut-google-drive-documents' ) )
+			);
+		}
+		// Order files by
+		else if ( 'sgdd_order_by' === $this->id ) {
+			$inputs = array(
+				array( 'name_asc', __( 'Name (ascending)', 'skaut-google-drive-documents' ) ),
+				array( 'name_dsc', __( 'Name (descending)', 'skaut-google-drive-documents' ) ),
+				array( 'time_asc', __( 'Time (ascending)', 'skaut-google-drive-documents' ) ),
+				array( 'time_dsc', __( 'Time (descending)', 'skaut-google-drive-documents' ) )
+			);
+		}
```

Comment:
Yes - it is ugly but much better anyway than previous code :) I can improve this but not in this PR - I already done some modification in base class of these settings and it will be easier to modify it after applying these modifications. We can mark this somehow as TODO work.

Reply:
Hmm, ok. If this were my codebase with my standards, I would say no (because a promise to fix bad code later is almost never fulfilled :D ). But given the really bad code quality here, I've filed it as #70 and let's solve it later.